### ISSUE TITLE
OP-TEE not working on STM32MP157A-DK1

### ISSUE THREAD

**Issue by f1bu** [OP-TEE/optee_os#5422]
Hi,

I tried to install OP-TEE on my STM32MP157A-DK1 board, which should be supported by OP-TEE, but I encountered some problems.
To come straight to my problem: the OP-TEE-OS and the Linux both seem to boot fine. But when I try to run xtest I always get the error message "_ERR [133] TSUP:main:819: failed to find an OP-TEE supplicant device_"

**How I did the build**
I built on an Ubuntu 20.04 system, following the build instructions on
https://optee.readthedocs.io/en/latest/building/devices/stm32mp1.html
and https://optee.readthedocs.io/en/latest/building/gits/build.html#get-and-build-the-solution
My commands:
`repo init -u https://github.com/OP-TEE/manifest.git -m stm32mp1.xml `
`repo sync -j4 --no-clone-bundle`
``make -j  `nproc` PLATFORM=stm32mp1-157A_DK1 all``

**Additional debugging information**
I attached my bootlog.

You can see the OP-TEE banner
`I/TC: OP-TEE version: 3.17.0-195-g5c64ea9c (gcc version 10.2.1 20201103 (GNU Toolchain for the A-profile Architecture 10.2-2020.11 (arm-10.16)))`

At the end you can also see the lines
```
Set permissions on /dev/tee*: chown: /dev/teepriv0: No such file or directory
FAIL
```
which makes sense, because there really is no tee* node in /dev

I would be happy about any help.

Regards
Finn

**Comment by jforissier** [comment]
Hello @f1bu,

Missing `optee` node in the DT perhaps?
```
	firmware {
		optee: optee {
			compatible = "linaro,optee-tz";
			method = "smc";
			status = "disabled";
		};
	};
```

**Comment by etienne-lms** [comment]
Indeed, optee reference distrib is using a linux kenrel fork that patches stm32mp1 DTS file to define and enable OP-TEE but only on EV1 and DK2 board, not for the DK1. You'll need to following patch in linux kernel tree (based on [linaro-swg/linux branch _optee_](https://github.com/linaro-swg/linux/commits/688e13588ae54c2c87096ba1fadea3b0d94f34cc)).
```patch
--- a/arch/arm/boot/dts/stm32mp157a-dk1.dts
+++ b/arch/arm/boot/dts/stm32mp157a-dk1.dts
@@ -15,6 +15,13 @@ / {
 	model = "STMicroelectronics STM32MP157A-DK1 Discovery Board";
 	compatible = "st,stm32mp157a-dk1", "st,stm32mp157";
 
+	reserved-memory {
+		optee_memory: optee@0xde000000 {
+			reg = <0xde000000 0x02000000>;
+			no-map;
+		};
+	};
+
 	aliases {
 		ethernet0 = &ethernet0;
 		serial0 = &uart4;
@@ -26,3 +33,7 @@ chosen {
 		stdout-path = "serial0:115200n8";
 	};
 };
+
+&optee {
+	status = "okay";
+};
```

**Comment by f1bu** [comment]
Hi @jforissier ,

Wow that was a quick answer :)

The kernel devicetree already had a node 

```
firmware {
		optee: optee {
			compatible = "linaro,optee-tz";
			method = "smc";
			status = "disabled";
		};
};
```

I removed the disabled-status line and now it just works!
Thank you very much for the hint, but I'm wondering why optee would be disabled for an optee build?

Best regards 
Finn

**Comment by f1bu** [comment]
Hi @etienne-lms ,

thanks for clarifying this. Thank you both for your quick support, I didn't expect to get a (helpful) answer that quick.

Regards,
Finn
